server/plugin.go

```diff
@@ -225,68 +230,84 @@ func (p *Plugin) getOAuthConfigForChimeraApp(scopes []string, redirectURL string
 }
 
 func (p *Plugin) storeGitlabUserInfo(info *gitlab.UserInfo) error {
+	jsonInfo, err := json.Marshal(info)
+	if err != nil {
+		return err
+	}
+
+	if err := p.API.KVSet(info.UserID+GitlabUserInfoKey, jsonInfo); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Plugin) storeGitlabUserToken(userID string, token *oauth2.Token) error {
 	config := p.getConfiguration()
 
-	encryptedToken, err := encrypt([]byte(config.EncryptionKey), info.Token.AccessToken)
+	encryptedToken, err := encrypt([]byte(config.EncryptionKey), token.AccessToken)
 	if err != nil {
 		return err
 	}
 
-	info.Token.AccessToken = encryptedToken
+	token.AccessToken = encryptedToken
```

Comment:
Yes, definitely

Reply:
Alright, I just committed this. Now, it's harder to test since you can't edit the expiration in the database. You can instead test by adding `tok.Expiry = time.Now()` in `completeConnectUserToGitlab` before the new token is saved. Then, the token refresh will be triggered by the app as it refreshes at the end of the connect user operation.